// File generated by `org.jetbrains.rhizomedb.TestGeneratorKt`. DO NOT MODIFY MANUALLY
import com.jetbrains.rhizomedb.*

@GeneratedEntityType
enum class MyEntity(override val eid: EID) : Entity {
    <!WRONG_ANNOTATION_TARGET!>@GeneratedEntityType<!>
    A(0),
    B(1);
}

fun foo() {
    MyEntity.<!UNRESOLVED_REFERENCE!>all<!>()
    MyEntity.<!UNRESOLVED_REFERENCE!>single<!>()
    MyEntity.<!UNRESOLVED_REFERENCE!>singleOrNull<!>()
}
