// File generated by `org.jetbrains.rhizomedb.TestGeneratorKt`. DO NOT MODIFY MANUALLY
import com.jetbrains.rhizomedb.*

interface BaseInterface

interface DerivedInterface : BaseInterface

<!NOT_ENTITY!>@GeneratedEntityType
data class MyEntity(val eid: EID) : DerivedInterface<!>

fun foo() {
    MyEntity.<!NONE_APPLICABLE!>all<!>()
    MyEntity.<!NONE_APPLICABLE!>single<!>()
    MyEntity.<!NONE_APPLICABLE!>singleOrNull<!>()
}
